Library/Homebrew/github_runner_matrix.rb

```diff
@@ -129,6 +129,23 @@ def runner_enabled?(macos_version)
     macos_version <= NEWEST_HOMEBREW_CORE_MACOS_RUNNER && macos_version >= OLDEST_HOMEBREW_CORE_MACOS_RUNNER
   end
 
+  NEW_INTEL_MACOS_MUST_BUILD_FORMULAE = %w[pkg-config pkgconf].freeze
+
+  sig { returns(T::Boolean) }
+  def deploy_new_x86_64_runner?
+    return true if @testing_formulae.any? { |f| NEW_INTEL_MACOS_MUST_BUILD_FORMULAE.include?(f.name) }
+    return true if @testing_formulae.any? { |f| f.formula.class.pour_bottle_only_if == :clt_installed }
+
+    Formula.all.any? do |formula|
+      non_test_dependencies = Dependency.expand(formula, cache_key: "determine-test-runners") do |_, dependency|
+        Dependency.prune if dependency.test?
```

Comment:
I don't think the unbottled check works this way. Even test dependencies get bottled (though there's basically no test-only dependencies in the affected trees anyway).

Reply:
It's how it's meant to work. If that's not how it actually works then that's a different story:

https://github.com/Homebrew/homebrew-test-bot/blob/e84b3ddb0821cd0145f1b8648705813df3c8e2bd/lib/tests/formulae.rb#L304-L316

Reply:
Ah I see what mistake I made - didn't realise `libgccjit` has a Intel only dependency.

Which is weird but whatever.